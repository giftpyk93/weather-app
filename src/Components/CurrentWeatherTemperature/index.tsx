import React, { FC } from "react";
import moment from "moment";
import Grid from "@material-ui/core/Grid";
import styled from "styled-components";

import { DATE_FORMAT } from "../../Constants/DateTimeFormat";
import { WeatherIcon } from "../Icon";

type CurrentWeatherPropsType = {
  currentTemp: number;
  iconCode: string;
  place: string;
  maxTemp: number;
  minTemp: number;
  weatherStatus: string;
};

const LocationText = styled.p`
  font-size: 1.5rem;
  font-weight: bold;
`;

const DateText = styled.p`
  font-size: 1.2rem;
`;

const CurrentTempText = styled.p`
  font-size: 4rem;
  font-weight: bold;
`;

const CurrentWeather: FC<CurrentWeatherPropsType> = props => {
  const { currentTemp, iconCode, place, maxTemp, minTemp, weatherStatus } = props;
  const currentDate = moment().format(DATE_FORMAT);
  return (
    <Grid container direction="column" className="p-3">
      <Grid item xs={12}>
        <LocationText>{place}</LocationText>
        <DateText>{currentDate}</DateText>
        <p>
          MIN {minTemp}°C, MAX {maxTemp}°C
        </p>
      </Grid>
      <Grid item container xs={12} direction="column" alignItems="center">
        <Grid item>
          <WeatherIcon iconCode={iconCode} />
        </Grid>
        <Grid item>
          <CurrentTempText>{currentTemp}°C</CurrentTempText>
        </Grid>
        <Grid item>
          <p>{weatherStatus}</p>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default CurrentWeather;
