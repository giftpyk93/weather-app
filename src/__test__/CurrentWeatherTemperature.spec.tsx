import React from "react";
import { create } from "react-test-renderer";
import CurrentWeatherTemperature from "Components/CurrentWeatherTemperature";

const mockCurrentWeatherTemperatureProp = {
  currentTemp: 27.48,
  iconCode: "d10",
  place: "Queensland, Australia",
  maxTemp: 27.48,
  minTemp: 27.48,
  weatherStatus: "Clear",
};

describe("CurrentWeatherTemperature component", () => {
  test("Matches the snapshot", () => {
    const renderer = create(<CurrentWeatherTemperature {...mockCurrentWeatherTemperatureProp} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
