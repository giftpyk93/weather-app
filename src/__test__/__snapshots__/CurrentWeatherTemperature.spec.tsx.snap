// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherTemperature component Matches the snapshot 1`] = `
<div
  className="MuiGrid-root p-3 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <p
      className="sc-bwzfXH eLhiJr"
    >
      Queensland, Australia
    </p>
    <p
      className="sc-htpNat buCuQu"
    >
      Saturday, 23 Nov 2019
    </p>
    <p>
      MIN 
      27.48
      °C, MAX 
      27.48
      °C
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <img
        alt="weather-icon"
        className="sc-bdVaJa cElyJo"
        size="big"
        src="http://openweathermap.org/img/wn/d10@2x.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="sc-bxivhb dPxHrp"
      >
        27.48
        °C
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p>
        Clear
      </p>
    </div>
  </div>
</div>
`;
